### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 15

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#255] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152315703-0bd42557-b2f7-468d-9427-945db9c2d866.png)
![image](https://user-images.githubusercontent.com/69389461/152315759-83602a4c-e86e-4860-9ad9-5ae49b26dae8.png)

**Comment by LuginiAndrea** [comment]
Per definzione, la funzione è inversa è la funzione (in questo caso chiamiamola $g$) tale che 
$$Y = f\left(X\right) \Longleftrightarrow X = g\left(Y\right)$$
Ovvero che, data l'immagine di $x \in X$ per $f$, ci restituisce $x$
Possiamo dedurre quindi 
$$\left(g \circ f\right)\left(X\right) = g\left(f\left(X\right)\right) = g\left(Y\right) = X \textrm{ per definizione }$$
Quindi posta $g = f^{-1} \Longrightarrow \left(g \circ f\right)\left(X\right)= X$. Il punto 1 implica il punto 2.
Possiamo anche dedurre che, prendendo per vera l'ipotesi 
$$\left(f \circ g \right)\left(Y\right) = Y$$
Questa implica 
$$f\left(g\left(Y\right)\right) = Y \Longrightarrow f\left(g\left(Y\right)\right) = f\left(X\right) \Longrightarrow g\left(Y\right) = X \Longrightarrow g = f^{-1}$$
Il punto 3 implica il punto 1.
Di conseguenza per implicazione circolare $$p_1 \Longrightarrow p_2 \wedge p_3 \Longrightarrow p_1 \Longleftrightarrow p_1 \Longleftrightarrow p_2 \wedge p_2 \Longleftrightarrow p_3 \wedge p_3 \Longleftrightarrow p_1$$
